purchases/src/main/kotlin/com/revenuecat/purchases/CustomerInfoHelper.kt

```diff
@@ -3,21 +3,48 @@ package com.revenuecat.purchases
 import android.os.Handler
 import android.os.Looper
 import com.revenuecat.purchases.common.Backend
+import com.revenuecat.purchases.common.DateProvider
+import com.revenuecat.purchases.common.DefaultDateProvider
 import com.revenuecat.purchases.common.LogIntent
+import com.revenuecat.purchases.common.between
 import com.revenuecat.purchases.common.caching.DeviceCache
 import com.revenuecat.purchases.common.debugLog
+import com.revenuecat.purchases.common.diagnostics.DiagnosticsTracker
 import com.revenuecat.purchases.common.errorLog
 import com.revenuecat.purchases.common.log
 import com.revenuecat.purchases.common.offlineentitlements.OfflineEntitlementsManager
 import com.revenuecat.purchases.interfaces.ReceiveCustomerInfoCallback
 import com.revenuecat.purchases.strings.CustomerInfoStrings
+import java.util.Date
+import kotlin.time.Duration
 
+/**
+ * Wrapper for [ReceiveCustomerInfoCallback] to hold extra information for diagnostics.
+ */
+internal class ReceiveCustomerInfoFullCallback(
+    private val onCustomerInfoReceived: (CustomerInfo, Boolean?) -> Unit,
+    private val onError: (PurchasesError, Boolean?) -> Unit,
+) : ReceiveCustomerInfoCallback {
+    var hadUnsyncedPurchases: Boolean? = null
```

Comment:
Hmm this pattern of a callback with state seems weird to me... I think I would prefer to keep this state samewhere else... Maybe we can make the `CustomerInfoHelper` return a different data type that contains all the data we need, and we just map that into the `ReceiveCustomerInfoCallback` in the `PurchasesOrchestrator`. This would allow us to remove the `ReceiveCustomerInfoFullCallback` as well. Wdyt?